enterprise/cmd/frontend/internal/auth/init.go

```diff
@@ -60,60 +61,68 @@ func Init(logger log.Logger, db database.DB) {
 	app.RegisterSSOSignOutHandler(ssoSignOutHandler)
 
 	// Warn about usage of auth providers that are not enabled by the license.
-	graphqlbackend.AlertFuncs = append(graphqlbackend.AlertFuncs, func(args graphqlbackend.AlertFuncArgs) []*graphqlbackend.Alert {
-		// Only site admins can act on this alert, so only show it to site admins.
-		if !args.IsSiteAdmin {
-			return nil
-		}
+	graphqlbackend.AlertFuncs = append(graphqlbackend.AlertFuncs, requireLicenseOrSuggestSSOAlerts)
+}
+
+func requireLicenseOrSuggestSSOAlerts(args graphqlbackend.AlertFuncArgs) []*graphqlbackend.Alert {
+	// Only site admins can act on this alert, so only show it to site admins.
+	if !args.IsSiteAdmin {
+		return nil
+	}
 
-		if licensing.IsFeatureEnabledLenient(licensing.FeatureSSO) {
-			return nil
+	collected := make(map[string]struct{})
+	var names []string
+	for _, p := range conf.Get().AuthProviders {
+		// Only built-in authentication provider is allowed by default.
+		if p.Builtin != nil {
+			continue
 		}
 
-		collected := make(map[string]struct{})
-		var names []string
-		for _, p := range conf.Get().AuthProviders {
-			// Only built-in authentication provider is allowed by default.
-			if p.Builtin != nil {
-				continue
-			}
-
-			var name string
-			switch {
-			case p.Github != nil:
-				name = "GitHub OAuth"
-			case p.Gitlab != nil:
-				name = "GitLab OAuth"
-			case p.Bitbucketcloud != nil:
-				name = "Bitbucket Cloud OAuth"
-			case p.AzureDevOps != nil:
-				name = "Azure DevOps"
-			case p.HttpHeader != nil:
-				name = "HTTP header"
-			case p.Openidconnect != nil:
-				name = "OpenID Connect"
-			case p.Saml != nil:
-				name = "SAML"
-			default:
-				name = "Other"
-			}
-
-			if _, ok := collected[name]; !ok {
-				collected[name] = struct{}{}
-				names = append(names, name)
-			}
+		var name string
+		switch {
+		case p.Github != nil:
+			name = "GitHub OAuth"
+		case p.Gitlab != nil:
+			name = "GitLab OAuth"
+		case p.Bitbucketcloud != nil:
+			name = "Bitbucket Cloud OAuth"
+		case p.AzureDevOps != nil:
+			name = "Azure DevOps"
+		case p.HttpHeader != nil:
+			name = "HTTP header"
+		case p.Openidconnect != nil:
+			name = "OpenID Connect"
+		case p.Saml != nil:
+			name = "SAML"
+		default:
+			name = "Other"
 		}
-		if len(names) == 0 {
-			return nil
+
+		if _, ok := collected[name]; !ok {
+			collected[name] = struct{}{}
+			names = append(names, name)
 		}
+	}
 
+	if len(names) > 0 && !licensing.IsFeatureEnabledLenient(licensing.FeatureSSO) {
```

Comment:
```suggestion
	names := collected.Values()
	if len(names) > 0 && !licensing.IsFeatureEnabledLenient(licensing.FeatureSSO) {
```

This way, we keep track of the names in one data structure instead of two.

Reply:
Thank you, updated a6ca4bcf1ae29b77d677e8a226c43260c3de24a8.